Missing dependency jspecify version attribute in parent pom

## Description

We noticed that the jspecify version entry is missing, causing publishing problems with jbang. There is no definition for the version at the top.

```

<dependency>
<groupId>org.jspecify</groupId>
<artifactId>jspecify</artifactId>
<version>${jspecify.version}</version>
</dependency>

```

## Configuration

**Version:**  2.25

**Operating system:**  macos/linux 

**JDK:**  24

## Logs

```
Error:  [ERROR] Could not resolve dependencies: The following artifacts could not be resolved: org.jspecify:jspecify:jar:${jspecify.version} (absent): 
Could not find artifact org.jspecify:jspecify:jar:${jspecify.version} in mavencentral (https://repo1.maven.org/maven2/)

```

## Reproduction

[An isolated test reproducing the test.
JUnit tests similar to the ones in the code base are extremely appreciated.]

--- a/log4j-parent/pom.xml
+++ b/log4j-parent/pom.xml
@@ -108,6 +108,7 @@
     <jeromq.version>0.6.0</jeromq.version>
     <jmdns.version>3.6.1</jmdns.version>
     <jmh.version>1.37</jmh.version>
+    <jspecify.version>1.0.0</jspecify.version>
     <junit.version>4.13.2</junit.version>
     <junit-jupiter.version>5.13.1</junit-jupiter.version>
     <junit-pioneer.version>1.9.1</junit-pioneer.version>
@@ -1141,6 +1142,53 @@
   </build>
 
   <profiles>
+
+    <!-- `java8-tests` profile to force using Java 8 while running tests -->
+    <profile>
+
+      <id>java8-tests</id>
+
+      <activation>
+        <property>
+          <name>env.CI</name>
+          <value>true</value>
+        </property>
+      </activation>
+
+      <!-- There are certain Java 8 bugs[1] that cause Mockito failures[2].
+           Adding necessary dependencies to workaround them.
+           [1] https://bugs.openjdk.org/browse/JDK-8152174
+           [2] https://github.com/mockito/mockito/issues/1449 -->
+      <dependencies>
+        <dependency>
+          <groupId>org.jspecify</groupId>
+          <artifactId>jspecify</artifactId>
+          <version>${jspecify.version}</version>
+          <scope>test</scope>
+        </dependency>
+      </dependencies>
+
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <executions>
+              <!-- Modifies only the `default-test` run -->
+              <execution>
+                <id>default-test</id>
+                <configuration>
+                  <jdkToolchain>
+                    <version>[1.8, 9)</version>
+                  </jdkToolchain>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+
+    </profile>
 
     <!-- `plugin-processing` profile containing configuration specific to `@Plugin`-annotated members -->
     <profile>

--- a/pom.xml
+++ b/pom.xml
@@ -357,7 +357,6 @@
     <!-- =====================================================
          Direct dependency version properties (in alphabetical order)
          ===================================================== -->
-    <jspecify.version>1.0.0</jspecify.version>
     <log4j-docgen.version>0.9.0</log4j-docgen.version>
     <node.version>21.7.1</node.version>
     <npm.version>10.5.0</npm.version>
@@ -955,53 +954,6 @@
   </build>
 
   <profiles>
-
-    <!-- `java8-tests` profile to force using Java 8 while running tests -->
-    <profile>
-
-      <id>java8-tests</id>
-
-      <activation>
-        <property>
-          <name>env.CI</name>
-          <value>true</value>
-        </property>
-      </activation>
-
-      <!-- There are certain Java 8 bugs[1] that cause Mockito failures[2].
-           Adding necessary dependencies to workaround them.
-           [1] https://bugs.openjdk.org/browse/JDK-8152174
-           [2] https://github.com/mockito/mockito/issues/1449 -->
-      <dependencies>
-        <dependency>
-          <groupId>org.jspecify</groupId>
-          <artifactId>jspecify</artifactId>
-          <version>${jspecify.version}</version>
-          <scope>test</scope>
-        </dependency>
-      </dependencies>
-
-      <build>
-        <plugins>
-          <plugin>
-            <groupId>org.apache.maven.plugins</groupId>
-            <artifactId>maven-surefire-plugin</artifactId>
-            <executions>
-              <!-- Modifies only the `default-test` run -->
-              <execution>
-                <id>default-test</id>
-                <configuration>
-                  <jdkToolchain>
-                    <version>[1.8, 9)</version>
-                  </jdkToolchain>
-                </configuration>
-              </execution>
-            </executions>
-          </plugin>
-        </plugins>
-      </build>
-
-    </profile>
 
     <profile>
       <id>rerun-tests</id>

--- a/src/changelog/.2.x.x/3758_fix_jspecify_dep.xml
+++ b/src/changelog/.2.x.x/3758_fix_jspecify_dep.xml
@@ -1,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<entry xmlns="https://logging.apache.org/xml/ns"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="
+           https://logging.apache.org/xml/ns
+           https://logging.apache.org/xml/ns/log4j-changelog-0.xsd"
+       type="fixed">
+    <issue id="3758" link="https://github.com/apache/logging-log4j2/issues/3758"/>
+    <description format="asciidoc">
+        Move `jspecify.version` Maven property and `java8-tests` profile from `log4j-bom` to `log4j-parent`, since the former gets trimmed before deployment.
+    </description>
+</entry>

